### PR TITLE
Adding pulp data transfer capability

### PR DESCRIPTION
**Pull request by johnpmitsch**
Adding the ability to transfer pulp_data file when cloning a satellite

### CODE REVIEW

File: roles/sat6repro/tasks/main.yml [hunk 1]
```diff
@@ -103,6 +105,6 @@
   file: src=/opt/rh/ruby193/root/usr/share/gems/gems/katello-2.2.0.93/public/assets/bastion_katello dest=/usr/share/foreman/public/assets/bastion_katello
   when: satelliteversion == 6.1
 
-- name: Reset katello index (Note: this might take hours!!!!!)
```
Comment (johnpmitsch) [inline roles/sat6repro/tasks/main.yml]: had issues with the `:`

Comment (sthirugn) [inline roles/sat6repro/tasks/main.yml]: Fixed with https://github.com/RedHatSatellite/satellite-clone/pull/25

File: roles/sat6repro/tasks/main.yml [hunk 2]
```diff
@@ -52,6 +52,11 @@
   yum: name=satellite state=latest
   when: satelliteversion == 6.2
 
+- name: "Install rsync on the Satellite"
```
Comment (sthirugn) [inline roles/sat6repro/tasks/main.yml]: s/Satellite/Ansible Control Node

File: roles/sat6repro/tasks/main.yml [hunk 3]
```diff
@@ -71,20 +82,11 @@
   command: katello-installer
   when: satelliteversion == 6.1
 - name: run Satellite 6.2 installer
-  command: satellite-installer --scenario satellite
+  command: satellite-installer --scenario satellite --foreman-proxy-dns false --foreman-proxy-dhcp false
```
Comment (sthirugn) [inline roles/sat6repro/tasks/main.yml]: what is the need to specify `--foreman-proxy-dns false --foreman-proxy-dhcp false` explicitly?

File: roles/sat6repro/tasks/main.yml [hunk 4]
```diff
@@ -103,6 +105,9 @@
   file: src=/opt/rh/ruby193/root/usr/share/gems/gems/katello-2.2.0.93/public/assets/bastion_katello dest=/usr/share/foreman/public/assets/bastion_katello
   when: satelliteversion == 6.1
 
+- include: reset_pulp_data.yml
```
Comment (sthirugn) [inline roles/sat6repro/tasks/main.yml]: Can you add some comments explaining why we need to do this?

File: roles/sat6repro/vars/main.yml [hunk 5]
```diff
@@ -18,11 +18,17 @@ config_files_tar_file_name: config_files.tar.gz
...
+include_pulp_data: false
+pulp_data_tar_file_name: pulp_data.tar
+
 
 # Registration info - Credentials to register your host to RH portal
-activationkey: ak1
+activationkey: changeme
 org: 111111
-
+pool: 111111
```
Comment (sthirugn) [inline roles/sat6repro/vars/main.yml]: why do we need this? this is not being used anywhere?

File: roles/sat6repro/tasks/main.yml (hunk 3 above)
Comment (johnpmitsch) [inline roles/sat6repro/tasks/main.yml]: having these on resulted in an error:

```[ERROR 2016-12-13 16:17:36 main]  Could not get the ip address from fact ipaddress_virbr1 at /usr/share/foreman-installer/modules/foreman_proxy/manifests/proxydhcp.pp:8 on node sat-r220-09.lab.eng.rdu2.redhat.com```

If the user wants to test provisioning, they are going to have to turn these back on

Comment (johnpmitsch) [inline roles/sat6repro/tasks/main.yml]: I'll make a note of this in the README

File: roles/sat6repro/tasks/main.yml (hunk 4 above)
Comment (johnpmitsch) [inline roles/sat6repro/tasks/main.yml]: If you clone w/o pulp_data file, the entries for rpms in repositories will still be in the pulp db. So pulp will think everything is synced, and wont re-sync new content. Also, the package counts will be wrong.

The steps in this yml file clean the pulp database, returning everything to an "empty, unsynced" state.

File: roles/sat6repro/vars/main.yml (hunk 5 above)
Comment (johnpmitsch) [inline roles/sat6repro/vars/main.yml]: I've been using it for an activation key, but probably unnecessarily...  I can remove it.

File: roles/sat6repro/tasks/main.yml (hunk 2 above)
Comment (johnpmitsch) [inline roles/sat6repro/tasks/main.yml]: will correct

**Pull request merged**